provers/groth16/arkworks-adapter/src/lib.rs

```diff
@@ -0,0 +1,90 @@
+#[cfg(test)]
+mod integration_tests;
+
+use ark_ff::PrimeField;
+use ark_relations::r1cs::{ConstraintSystemRef, Field};
+use lambdaworks_groth16::{common::*, r1cs::R1CS, QuadraticArithmeticProgram};
+use lambdaworks_math::traits::ByteConversion;
+
+use std::ops::Deref;
+
+pub fn to_lambda<F: PrimeField>(
+    cs: &ConstraintSystemRef<F>,
+) -> (QuadraticArithmeticProgram, Vec<FrElement>) {
+    (
+        QuadraticArithmeticProgram::from_r1cs(r1cs_from_arkworks_cs(cs)),
+        extract_witness_from_arkworks_cs(cs),
+    )
+}
+
+fn r1cs_from_arkworks_cs<F: PrimeField>(cs: &ConstraintSystemRef<F>) -> R1CS {
+    cs.inline_all_lcs();
+
+    let r1cs_matrices = cs.to_matrices().unwrap();
+    let num_pub_vars = cs.num_instance_variables();
+    let total_variables = cs.num_witness_variables() + num_pub_vars - 1;
+
+    R1CS::from_matrices(
+        ark_to_lambda_matrix(&r1cs_matrices.a, total_variables),
+        ark_to_lambda_matrix(&r1cs_matrices.b, total_variables),
+        ark_to_lambda_matrix(&r1cs_matrices.c, total_variables),
+        num_pub_vars - 1,
+        0,
+    )
+}
+
+fn extract_witness_from_arkworks_cs<F: PrimeField>(cs: &ConstraintSystemRef<F>) -> Vec<FrElement> {
+    let binding = cs.borrow().unwrap();
+    let borrowed_cs_ref = binding.deref();
+
+    let ark_witness = &borrowed_cs_ref.witness_assignment;
+    let ark_io = &borrowed_cs_ref.instance_assignment[1..].to_vec();
```

Comment:
Comment why is it 1, so it's not a magical number

Reply:
It was unnecessary. Removed.